planner/core/planbuilder.go

```diff
@@ -411,13 +411,37 @@ const (
 	renameView
 )
 
+type cteInfo struct {
```

Comment:
can we extract a common part for it and `CTEClass`? The fields in `CTEClass` are almost found in the `cteInfo`.

Reply:
I think it's not necessary, it makes the structure complex.

Reply:
We comment on most fields for this structure. Or we can comment on the `CTEClass` that we can find what each field means here.